### DISCUSSION TITLE
Cannot read properties of null (reading 'registered') - Next.js

### DISCUSSION THREAD

**Discussion by zgrybus** [chakra-ui/chakra-ui#8521] [Q&A] [answered]
### Description

Hi!

I use `@chakra-ui/react` (v2.8.2) inside my Next.js App this way ( and this is working fine! ):

```typescript
import { Flex } from '@chakra-ui/react';

const MyNewComponent = () => (
 <Flex>Wonderful component</Flex>
)
```

But when I try to create totally new component's library that uses `@chakra-ui/react` ( basically my component's library will export this `MyNewComponent` component ) and use it inside Next.js application I'm getting this error

```typescript
// components_library/main.ts of the component's library
export { Flex } from '@chakra-ui/react';

// next_js_app/src/components/Component.ts file where I use components library
import { Flex } from 'components-library';

const MyNewComponent = () => (
 <Flex>Wonderful component</Flex>
)
```

Server Error
**TypeError: Cannot read properties of null (reading 'registered')**

This error happened while generating the page. Any console logs will be displayed in the terminal window.

Other components that used external libraries are working fine. The problem is only with `@chakra-ui/react` library.

Any ideas? 

Basically, the error happens only when I bundle the `@chakra-ui/react` 🤔 
This is vite.config.ts of my components library
```typescript
import path from 'node:path';

import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';
import tsconfigPaths from 'vite-tsconfig-paths';

export default defineConfig({
  plugins: [
    tsconfigPaths(),
    dts({
      exclude: ['packages/lib/**/*.stories.tsx'],
      insertTypesEntry: true,
    }),
  ],
  build: {
    lib: {
      entry: path.resolve(__dirname, './packages/lib/main.tsx'),
      name: 'components-library',
      formats: ['es'],
    },
    emptyOutDir: true,
    rollupOptions: {
      external: [
        'react',
        'react-dom'
      ],
      output: {
        globals: {
          react: 'React',
          'react-dom': 'ReactDOM',
        },
      },
    },
  },
});

```

**This issue is similar I guess 
https://github.com/JedWatson/react-select/issues/5555**

### Link to Reproduction

not provided

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### Chakra UI Version

2.8.2

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

**Comment by zgrybus** [comment]
@segunadebayo any help?

**Comment by Yeonny0723** [comment]
Same issue here, got any solution for this?

**Comment by Yeonny0723** [comment]
Hey, I think you can refer to discussion https://github.com/chakra-ui/chakra-ui/discussions/8804

**Comment by segunadebayo** [comment, marked as answer]
This is two copies of React / Emotion. The Vite library bundled `@chakra-ui/react` instead of leaving it external, so Next.js and the library each have their own Emotion cache.

Mark `@chakra-ui/react`, `@emotion/react`, `react`, and `react-dom` as peer deps and `rollupOptions.external`. If the host is Next.js App Router, dynamic-import your library provider with `ssr: false`.

Workaround from https://github.com/chakra-ui/chakra-ui/discussions/8804
